MetaMask/metamask-mobile
# fix: use hostname to fetch approvedhosts

## **Description**
This PR fixes an Android SDK bug whereby:
- The wallet was always returning the first permitted account instead of the selected account when connecting the SDK to MetaMask
- Changing the selected account on MetaMask did not fire a `metamask_accountsChanged` event so the SDK did not get selected account updated. 

This change implements the `getApprovedHosts` to fix this issue. 

## **Manual testing steps**

_1. Have more than one address in your metamask account and select any address as active address
_2. Connect from Android Metamask SDK and ensure the active account is the selected address on the SDK
_3. Switch to another account on the wallet, and ensure that the SDK updates to the new address

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
